Notebook llama (#739)
meta-llama/llama-recipes: recipes/quickstart/NotebookLlama/README.md

## NotebookLlama: An Open Source version of NotebookLM

[Listen to audio from the example here](./resources/_podcast.mp3)

This is a guided series of tutorials/notebooks that can be taken as a reference or course to build a PDF to Podcast workflow. 

> You will also learn from my experimentation of using Text to Speech Models.

Reviewer: ```suggestion
You will also learn from experimentations of using  Text to Speech Models.
```

It assumes zero knowledge of LLMs, prompting and audio models, everything is covered in their respective notebooks.

### Outline:

Here is step by step thought (pun intended) for the task:

> - Step 1: Pre-process PDF: Use `Llama-3.2-1B` to pre-process and save a PDF

Reviewer: ```suggestion
- Step 1: Pre-process PDF: Use `Llama-3.2-1B` to pre-process the PDF and save it in a `.txt` file.
```

> - Step 2: Transcript Writer: Use `Llama-3.1-70B` model to write a podcast transcript from the text

Reviewer: did 8B fail to write a good transcript?

Author: It's subjective since I didn't do any quantitative reviews but 70B felt more creative when doing the first parse. 

Should I change to 8B?

Author: Added a note

- Step 3: Dramatic Re-Writer: Use `Llama-3.1-8B` model to make the transcript more dramatic
- Step 4: Text-To-Speech Workflow: Use `parler-tts/parler-tts-mini-v1` and `bark/suno` to generate a conversational podcast

### Detailed steps on running the notebook:

> Requirements: GPU server or an API provider for using 70B, 8B and 1B Llama models.

Reviewer: maybe adding a more like for 70B ( GPUs with aggregated memory of >140 GB) etc.

Note: For our GPU Poor friends, you can also use the 8B and lower models for the entire pipeline. There is no strong recommendation. The pipeline below is what worked best on first few tests. You should try and see what works best for you!

- First, please Install the requirements from [here]() by running inside the folder:

```
git clone https://github.com/meta-llama/llama-recipes
cd llama-recipes/recipes/quickstart/NotebookLlama/

> pip install -r requirements.txt

Reviewer: requirements.txt is being missed here.

Author: Sorry-this is my mistake. Pushing a fix

```

- Notebook 1:

This notebook is used for processing the PDF and processing it using the new Feather light model into a `.txt` file.

Update the first cell with a PDF link that you would like to use. Please decide on a PDF to use for Notebook 1, it can be any link but please remember to update the first cell of the notebook with the right link. 

> Please try changing the prompts for the `Llama-3.2-1B-Instruct` model and see if you can improve results.

Reviewer: how do we ensure the LLM is not changing the text/ keep the fidelity of the original PDF?

Author: I will add it here in the ReadMe outline as well-please see the prompt for the 1B model. It's prompted to not change/summarise

- Notebook 2:

> This notebook will take in the processed output from Notebook 1 and creatively convert it into a podcast transcript using the `Llama-3.1-70B-Instruct` model. If you are GPU or even generally rich, please feel free to test with the 405B model!

Reviewer: ```suggestion
This notebook will take in the processed output from Notebook 1 and creatively convert it into a podcast transcript using the `Llama-3.1-70B-Instruct` model. If you are GPU rich, please feel free to test with the 405B model!
```
